pyproject.toml

```diff
@@ -23,8 +23,9 @@ maintainers = []
 keywords = []
 classifiers = [] # https://pypi.org/classifiers/
 dependencies = [
-  "grpcio ~= 1.53.0",
-  "grpcio_tools ~= 1.53.0"
+    "grpcio ~= 1.53.0",
+    "grpcio_tools ~= 1.53.0",
+    "pytest>=7.4.0",
```

Comment:
These are run-time deps, and `pytest` shouldn't be installed in the target environment, it's just a build-time dep. In `poetry` they had build/dev-deps (same as `npm` and `package.json`), but `pdm` only offers [a different approach (group deps)](https://daobook.github.io/pdm/usage/dependency/#add-development-only-dependencies).

Reply:
> These are run-time deps, and `pytest` shouldn't be installed in the target environment, it's just a build-time dep. In `poetry` they had build/dev-deps (same as `npm` and `package.json`), but `pdm` only offers [a different approach (group deps)](https://daobook.github.io/pdm/usage/dependency/#add-development-only-dependencies).

@epuertat thank you for all your help. We could split `builder-base` defined in the Dockerfile into `builder-base-dev` and `builder-base-release` and chain target/test/dev images accordingly? Any other more elegant approaches? I'll take a look into this issue in the separate PR, if you do not mind.